File: app/src/main/java/co/ostorlab/insecure_app/bugs/calls/StaticIV.java
package co.ostorlab.insecure_app.bugs.calls;

import android.util.Base64;
import android.util.Log;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

import co.ostorlab.insecure_app.BugRule;


public final class StaticIV extends BugRule {

    private static final String TAG = "RULE";

    @Override
    public String getDescription() {
        return "Use hardcoded static IV";
    }

    @Override
    public void run() throws Exception{
        byte[] IV = "0123456789abcdef".getBytes();
        String clearText = "Jan van Eyck was here 1434";
        String key = "ThisIs128bitSize";
        SecretKeySpec skeySpec = new SecretKeySpec(key.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
        cipher.init(Cipher.ENCRYPT_MODE, skeySpec, new IvParameterSpec(IV));
        byte[] encryptedMessage = cipher.doFinal(clearText.getBytes());
        Log.i(TAG, String.format("Message: %s", Base64.encodeToString(encryptedMessage, Base64.DEFAULT)));
    }
}

File: ostorlab_insecure_flutter_app/lib/bugs/static_iv.dart
import 'dart:convert';
import 'package:encrypt/encrypt.dart';
import 'package:ostorlab_insecure_flutter_app/bug_rule.dart';

/// A [BugRule] implementation that triggers the use of static IV for encryption.
class StaticIV extends BugRule {
  /// The tag used to identify instances of this rule.
  static const String _tag = "StaticIV";

  /// Returns a description of this [BugRule] implementation.
  @override
  String get description => 'Use of hardcoded static IV';

  /// Trigger the bug rule
  @override
  Future<void> run(String input) async {
    String data = "Jan van Eyck was here 1434";
    String password = "ThisIs128bitSize";

    final plainText = utf8.encode(data);
    final key = Key.fromUtf8(password);
    final iv = IV.fromUtf8("0123456789abcdef");
    final encrypter = Encrypter(AES(key, mode: AESMode.cbc));
    final encrypted = encrypter.encryptBytes(plainText, iv: iv);
    final encryptedString = base64.encode(encrypted.bytes);
    print(_tag + ' Message: ' + 'Encrypted data: $encryptedString');
  }
}
